### PR TITLE
block-storage: update names and labels of volumes and snapshots

### PR DESCRIPTION
**Pull request by sauterp**
# Description
Previously, changing block-storage volume/snapshot labels and names required recreating the resources and thereby deleting the data they held. We leverage the new `update-block-storage-{volume,snapshot}` operations update names and labels without recreating the udpated volumes or snapshots.

## Checklist
(For exoscale contributors)

* [x] Changelog updated (under *Unreleased* block)
* [x] Acceptance tests OK
* [x] For a new resource, datasource or new attributes: acceptance test added/updated

## Testing

New tests were added to verify that labels can be added, modified and also cleared. And the tests also verify that names of volumes/snapshots can be updated without modifying the labels or vice versa.

### CODE REVIEW

**Review by sauterp — COMMENTED** [review]
the implementation lgtm... but do we really need to add almost 1 million lines of vendored code???

File: pkg/resources/block_storage/resource_snapshot.go [hunk 1]
```diff
@@ -379,7 +379,112 @@ func (r *ResourceSnapshot) Read(ctx context.Context, req resource.ReadRequest, r
...
+		)
+		return
+	}
+
+	update := false
+	updateReq := exoscale.UpdateBlockStorageSnapshotRequest{}
+
+	if !plan.Name.Equal(state.Name) {
+		update = true
+
```
Comment (pierre-emmanuelJ) [inline pkg/resources/block_storage/resource_snapshot.go:425]:
```suggestion
```

**Review by pierre-emmanuelJ — APPROVED** [review]

File: pkg/resources/block_storage/resource_snapshot.go (hunk 1 above)
Comment (sauterp) [reply pkg/resources/block_storage/resource_snapshot.go:425]: I actually like this newline :)

File: docs/resources/block_storage_volume_snapshot.md [hunk 2]
```diff
@@ -20,13 +20,13 @@ Block Storage offers persistent externally attached volumes for your workloads.
...
 
-- `name` (String) ❗ Volume snapshot name.
+- `name` (String) Volume snapshot name.
 - `volume` (Attributes) Volume from which to create a snapshot. (see [below for nested schema](#nestedatt--volume))
 - `zone` (String) ❗ The Exoscale [Zone](https://www.exoscale.com/datacenters/) name.
 
 ### Optional
 
-- `labels` (Map of String) ❗ Resource labels. Not updateble after creation.
+- `labels` (Map of String) Resource labels. Not updateble after creation.
```
Comment (kobajagi) [inline docs/resources/block_storage_volume_snapshot.md]:
```suggestion
- `labels` (Map of String) Resource labels.
```

File: pkg/resources/block_storage/resource_snapshot.go [hunk 3]
```diff
@@ -105,11 +101,8 @@ func (r *ResourceSnapshot) Schema(ctx context.Context, req resource.SchemaReques
 			},
 			"labels": schema.MapAttribute{
 				ElementType:         types.StringType,
-				MarkdownDescription: "❗ Resource labels. Not updateble after creation.",
+				MarkdownDescription: "Resource labels. Not updateble after creation.",
```
Comment (kobajagi) [inline pkg/resources/block_storage/resource_snapshot.go]:
```suggestion
				MarkdownDescription: "Resource labels.",
```

**Review by kobajagi — APPROVED** [review]

**Pull request merged**